instrumentation/rocketmq-client-4.8/testing/src/main/groovy/io/opentelemetery/instrumentation/rocketmq/AbstractRocketMqClientTest.groovy

```diff
@@ -0,0 +1,190 @@
+            "messaging.rocketmq.tags" "TagA"
+            "messaging.rocketmq.broker_address" brokerAddr
+            "messaging.rocketmq.send_result" "SEND_OK"
+          }
+        }
+        span(2) {
+          name sharedTopic + " process"
+          kind CONSUMER
+          attributes {
+            "${SemanticAttributes.MESSAGING_SYSTEM.key}" "rocketmq"
+            "${SemanticAttributes.MESSAGING_DESTINATION.key}" sharedTopic
+            "${SemanticAttributes.MESSAGING_DESTINATION_KIND.key}" "topic"
+            "${SemanticAttributes.MESSAGING_OPERATION.key}" "process"
+            "${SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES.key}" Long
+            "${SemanticAttributes.MESSAGING_MESSAGE_ID.key}" String
+            "messaging.rocketmq.tags" "TagA"
+            "messaging.rocketmq.broker_address" brokerAddr
+            "messaging.rocketmq.queue_id" Long
+            "messaging.rocketmq.queue_offset" Long
+
+          }
+        }
+      }
+      cleanup:
+      producer.shutdown()
+      consumer.shutdown()
+      BaseConf.deleteTempDir()
+    }
+  }
+
+
+  def "test rocketmq produce and orderly consume"() {
+    setup:
+    producer = BaseConf.getProducer(BaseConf.nsAddr)
+    consumer = BaseConf.getConsumer(BaseConf.nsAddr, sharedTopic, "*", new RMQOrderListener())
+    when:
+    runUnderTrace("parent") {
+      producer.send(msg)
+    }
+    then:
+    assertTraces(1) {
+      trace(0, 3) {
+        basicSpan(it, 0, "parent")
+        span(1) {
+          name sharedTopic + " send"
+          kind PRODUCER
+          attributes {
+            "${SemanticAttributes.MESSAGING_SYSTEM.key}" "rocketmq"
+            "${SemanticAttributes.MESSAGING_DESTINATION.key}" sharedTopic
+            "${SemanticAttributes.MESSAGING_DESTINATION_KIND.key}" "topic"
+            "${SemanticAttributes.MESSAGING_MESSAGE_ID.key}" String
+            "messaging.rocketmq.tags" "TagA"
+            "messaging.rocketmq.broker_address" brokerAddr
+            "messaging.rocketmq.send_result" "SEND_OK"
+          }
+        }
+        span(2) {
+          name sharedTopic + " process"
+          kind CONSUMER
+          attributes {
+            "${SemanticAttributes.MESSAGING_SYSTEM.key}" "rocketmq"
+            "${SemanticAttributes.MESSAGING_DESTINATION.key}" sharedTopic
+            "${SemanticAttributes.MESSAGING_DESTINATION_KIND.key}" "topic"
+            "${SemanticAttributes.MESSAGING_OPERATION.key}" "process"
+            "${SemanticAttributes.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES.key}" Long
+            "${SemanticAttributes.MESSAGING_MESSAGE_ID.key}" String
+            "messaging.rocketmq.tags" "TagA"
+            "messaging.rocketmq.broker_address" brokerAddr
+            "messaging.rocketmq.queue_id" Long
+            "messaging.rocketmq.queue_offset" Long
+          }
+        }
+      }
+      cleanup:
+      producer.shutdown()
+      consumer.shutdown()
+      BaseConf.deleteTempDir()
+    }
+  }
+}
```

Comment:
Can you add some test cases that verify the batch message consumer scenario?

Reply:
Done